asset/file_reader/index.js

```diff
@@ -0,0 +1,108 @@
+'use strict';
+
+const Promise = require('bluebird');
+const Queue = require('@terascope/queue');
+const { getChunk, getOffsets } = require('@terascope/chunked-file-reader');
+const fse = require('fs-extra');
+
+function newSlicer(context, executionContext, retryData, logger) {
+    const opConfig = executionContext.config.operations[0];
+    const queue = new Queue();
+
+    async function processFile(filePath) {
+        const stat = await fse.stat(filePath);
+        const total = stat.size;
+        getOffsets(opConfig.size, total, opConfig.delimiter).forEach((offset) => {
+            offset.path = filePath;
+            queue.enqueue(offset);
+        });
+    }
+
+    async function getFilePaths(filePath) {
+        const dirContents = await fse.readdir(filePath);
+        return Promise.map(dirContents, async (file) => {
+            const fullPath = `${filePath}/${file}`;
+            const stats = await fse.lstat(fullPath);
+            if (stats.isFile()) {
+                return processFile(fullPath);
+            }
+            if (stats.isDirectory()) {
+                return getFilePaths(fullPath);
+            }
+            logger.error(`${file} is not a file or directory!!`);
+            return true;
+        })
+            .return([() => queue.dequeue()]);
+    }
+
+    return getFilePaths(opConfig.path)
+        .catch((err) => {
+            logger.error(err, 'Error while reading slicing files');
+            return Promise.reject(err);
```

Comment:
This should not log the error and reject it so that it doesn't get logged multiple times. I recommend using the TSError class for wrapping the error into with additional context. For example:

```js
const { TSError } = require('@terascope/utils');
// ...
    return getFilePaths(opConfig.path)
        .catch((err) => {
            const error = new TSErorr(err, {
                 reason:  'Error while reading slicing files'
            });
            return Promise.reject(err);
      });
// ...
```

Reply:
Replaced redundant error messages with the `TSError`